// tdesign-mobile-vue/cjs/_common/js/date-picker/format.js
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

require('lodash/isString');
require('dayjs');
var _common_js_datePicker_utils = require('../../../_chunks/dep-2c72af10.js');
require('../log/log.js');
require('../../../_chunks/dep-f98b59f6.js');
require('@babel/runtime/helpers/slicedToArray');
require('@babel/runtime/helpers/objectDestructuringEmpty');
require('@babel/runtime/helpers/extends');
require('@babel/runtime/helpers/defineProperty');
require('lodash/isFunction');
require('../../../_chunks/dep-35a4d9c4.js');
require('../../../_chunks/dep-7ba6caae.js');
require('lodash/chunk');



exports.TIME_FORMAT = _common_js_datePicker_utils.TIME_FORMAT;
exports.calcFormatTime = _common_js_datePicker_utils.calcFormatTime;
exports.formatDate = _common_js_datePicker_utils.formatDate;
exports.formatTime = _common_js_datePicker_utils.formatTime;
exports.getDefaultFormat = _common_js_datePicker_utils.getDefaultFormat;
exports.initYearMonthTime = _common_js_datePicker_utils.initYearMonthTime;
exports.isValidDate = _common_js_datePicker_utils.isValidDate;
exports.parseToDayjs = _common_js_datePicker_utils.parseToDayjs;
