// tdesign-mobile-vue/es/_common/js/date-picker/format.mjs
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

import '../../../_chunks/dep-ab2bdb8f.mjs';
import '../../../_chunks/dep-c48250ca.mjs';
export { T as TIME_FORMAT, u as calcFormatTime, t as formatDate, v as formatTime, w as getDefaultFormat, x as initYearMonthTime, r as isValidDate, q as parseToDayjs } from '../../../_chunks/dep-fcbf2697.mjs';
import '../log/index.mjs';
import '../log/log.mjs';
import '../../../_chunks/dep-57a75b98.mjs';
import '../../../_chunks/dep-c6268b4a.mjs';
import '../../../_chunks/dep-b617aa7d.mjs';
import '../../../_chunks/dep-ff449344.mjs';
import '../../../_chunks/dep-a9ebd588.mjs';
import '../../../_chunks/dep-06d482c6.mjs';
import '../../../_chunks/dep-4c2819a4.mjs';
import '../../../_chunks/dep-4b5beaed.mjs';
import '../../../_chunks/dep-dc33ef46.mjs';
import '../../../_chunks/dep-14bda5a6.mjs';
import '../../../_chunks/dep-c06e51c6.mjs';
import '../../../_chunks/dep-24878a92.mjs';
import '../../../_chunks/dep-654f8891.mjs';
import '../../../_chunks/dep-feb307dc.mjs';
import '../../../_chunks/dep-4a37904d.mjs';
import '../../../_chunks/dep-9f2b5648.mjs';
import '../../../_chunks/dep-ce88baea.mjs';
import '../../../_chunks/dep-4d5ede6c.mjs';
import '../../../_chunks/dep-5fe3ebe0.mjs';
import '../../../_chunks/dep-48081c7e.mjs';
import '../../../_chunks/dep-9f686842.mjs';
