// tdesign-mobile-vue/cjs/notice-bar/props.js
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var props = {
  content: {
    type: [String, Array, Function]
  },
  direction: {
    type: String,
    default: "horizontal",
    validator: function validator(val) {
      if (!val) return true;
      return ["horizontal", "vertical"].includes(val);
    }
  },
  marquee: {
    type: [Boolean, Object],
    default: false
  },
  operation: {
    type: [String, Function]
  },
  prefixIcon: {
    type: [Boolean, Function],
    default: true
  },
  suffixIcon: {
    type: Function
  },
  theme: {
    type: String,
    default: "info",
    validator: function validator(val) {
      if (!val) return true;
      return ["info", "success", "warning", "error"].includes(val);
    }
  },
  visible: {
    type: Boolean,
    default: void 0
  },
  modelValue: {
    type: Boolean,
    default: void 0
  },
  defaultVisible: Boolean,
  onClick: Function
};

exports["default"] = props;
