// tdesign-mobile-vue/cjs/textarea/props.js
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var props = {
  allowInputOverMax: Boolean,
  autofocus: Boolean,
  autosize: Boolean,
  bordered: Boolean,
  disabled: {
    type: Boolean,
    default: void 0
  },
  indicator: Boolean,
  label: {
    type: [String, Function]
  },
  layout: {
    type: String,
    default: "horizontal",
    validator: function validator(val) {
      if (!val) return true;
      return ["vertical", "horizontal"].includes(val);
    }
  },
  maxcharacter: {
    type: Number
  },
  maxlength: {
    type: Number
  },
  name: {
    type: String,
    default: ""
  },
  placeholder: {
    type: String,
    default: void 0
  },
  readonly: Boolean,
  value: {
    type: [String, Number],
    default: void 0
  },
  modelValue: {
    type: [String, Number],
    default: void 0
  },
  defaultValue: {
    type: [String, Number]
  },
  onBlur: Function,
  onChange: Function,
  onFocus: Function
};

exports["default"] = props;
