// tdesign-mobile-vue/cjs/steps/props.js
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var props = {
  current: {
    type: [String, Number],
    default: void 0
  },
  defaultCurrent: {
    type: [String, Number]
  },
  currentStatus: {
    type: String,
    default: "process",
    validator: function validator(val) {
      if (!val) return true;
      return ["default", "process", "finish", "error"].includes(val);
    }
  },
  layout: {
    type: String,
    default: "horizontal",
    validator: function validator(val) {
      if (!val) return true;
      return ["horizontal", "vertical"].includes(val);
    }
  },
  readonly: Boolean,
  theme: {
    type: String,
    default: "default",
    validator: function validator(val) {
      if (!val) return true;
      return ["default", "dot"].includes(val);
    }
  },
  sequence: {
    type: String,
    default: "positive",
    validator: function validator(val) {
      if (!val) return true;
      return ["positive", "reverse"].includes(val);
    }
  },
  onChange: Function
};

exports["default"] = props;
