// tdesign-mobile-vue/es/link/link.mjs
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

import { _ as _defineProperty } from '../_chunks/dep-dc33ef46.mjs';
import { defineComponent, computed, createVNode } from 'vue';
import config from '../config.mjs';
import props from './props.mjs';
import { useTNodeJSX, useContent } from '../hooks/tnode.mjs';
import { usePrefixClass } from '../hooks/useClass.mjs';
import { useFormDisabled } from '../form/hooks.mjs';
import '../_chunks/dep-b617aa7d.mjs';
import '../_chunks/dep-14bda5a6.mjs';
import '../_chunks/dep-57a75b98.mjs';
import '../_chunks/dep-c6268b4a.mjs';
import '../_chunks/dep-c06e51c6.mjs';
import '../_chunks/dep-ca4b618a.mjs';
import '../_chunks/dep-c0c7e354.mjs';
import '../_chunks/dep-12b97090.mjs';
import '../_chunks/dep-ff449344.mjs';
import '../_chunks/dep-a9ebd588.mjs';
import '../_chunks/dep-9f686842.mjs';
import '../_chunks/dep-06d482c6.mjs';
import '../_chunks/dep-feb307dc.mjs';
import '../_chunks/dep-f70b3591.mjs';
import '../_chunks/dep-82b3737e.mjs';
import '../hooks/render-tnode.mjs';
import '../_chunks/dep-35186489.mjs';
import '../_chunks/dep-afd6e78d.mjs';
import '../_chunks/dep-44fb3d82.mjs';
import '../_chunks/dep-ce88baea.mjs';
import '../_chunks/dep-4d5ede6c.mjs';
import '../_chunks/dep-32f76754.mjs';
import '../_chunks/dep-4f0b3d47.mjs';
import '../_chunks/dep-ab2bdb8f.mjs';
import '../_chunks/dep-ceb2af97.mjs';
import '../_chunks/dep-8b0eae2f.mjs';
import '../_chunks/dep-dffd3346.mjs';
import '../_chunks/dep-4a0b07f2.mjs';
import '../_chunks/dep-9f2b5648.mjs';
import '../_chunks/dep-dff870f8.mjs';
import '../_chunks/dep-8eaf2b7e.mjs';
import '../_chunks/dep-5fe3ebe0.mjs';
import '../_chunks/dep-0c24a69a.mjs';
import '../_chunks/dep-0d47707f.mjs';
import '../_chunks/dep-0bb5e0ba.mjs';
import '../_chunks/dep-4a37904d.mjs';
import '../_common/js/global-config/mobile/default-config.mjs';
import '../_common/js/global-config/mobile/locale/zh_CN.mjs';
import '../_chunks/dep-fbaa4502.mjs';
import '../_chunks/dep-c48250ca.mjs';
import '../_chunks/dep-8398b33d.mjs';
import '../config-provider/type.mjs';
import '../_chunks/dep-60e514fa.mjs';

var prefix = config.prefix;
var _Link = defineComponent({
  name: "".concat(prefix, "-link"),
  props: props,
  setup: function setup(props2) {
    var linkClass = usePrefixClass("link");
    var renderTNodeJSX = useTNodeJSX();
    var renderTNodeContent = useContent();
    var isDisabled = useFormDisabled();
    var linkClasses = computed(function () {
      return [linkClass.value, "".concat(linkClass.value, "--").concat(props2.theme || "default"), "".concat(linkClass.value, "--").concat(props2.size || "medium"), _defineProperty(_defineProperty(_defineProperty({}, "".concat(linkClass.value, "--disabled"), props2.disabled), "".concat(linkClass.value, "--underline"), props2.underline), "".concat(linkClass.value, "--hover"), props2.hover && !props2.disabled)];
    });
    var handleClick = function handleClick(e) {
      var _props2$onClick;
      if (props2.disabled) return;
      (_props2$onClick = props2.onClick) === null || _props2$onClick === void 0 || _props2$onClick.call(props2, e);
    };
    var renderContent = function renderContent() {
      var content = renderTNodeContent("default", "content");
      return content ? createVNode("span", {
        "class": ["".concat(linkClass.value, "__content")]
      }, [content]) : null;
    };
    var renderPrefixIcon = function renderPrefixIcon() {
      var prefixIcon = renderTNodeJSX("prefixIcon");
      return prefixIcon ? createVNode("span", {
        "class": ["".concat(linkClass.value, "__prefix-icon")]
      }, [prefixIcon]) : null;
    };
    var renderSuffixIcon = function renderSuffixIcon() {
      var suffixIcon = renderTNodeJSX("suffixIcon");
      return suffixIcon ? createVNode("span", {
        "class": ["".concat(linkClass.value, "__suffix-icon")]
      }, [suffixIcon]) : null;
    };
    return function () {
      return createVNode("a", {
        "class": linkClasses.value,
        "aria-disabled": isDisabled.value,
        "target": props2.target,
        "href": isDisabled.value || !props2.href ? void 0 : props2.href,
        "onClick": handleClick
      }, [renderPrefixIcon(), renderContent(), renderSuffixIcon()]);
    };
  }
});

export { _Link as default };
