// tdesign-mobile-vue/es/table/base-table-props.mjs
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

var baseTableProps = {
  bordered: Boolean,
  cellEmptyContent: {
    type: [String, Function]
  },
  columns: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  data: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  empty: {
    type: [String, Function],
    default: ""
  },
  fixedRows: {
    type: Array
  },
  height: {
    type: [String, Number]
  },
  loading: {
    type: [Boolean, Function],
    default: void 0
  },
  loadingProps: {
    type: Object
  },
  maxHeight: {
    type: [String, Number]
  },
  rowKey: {
    type: String,
    default: "id",
    required: true
  },
  showHeader: {
    type: Boolean,
    default: true
  },
  stripe: Boolean,
  tableContentWidth: {
    type: String,
    default: ""
  },
  tableLayout: {
    type: String,
    default: "fixed",
    validator: function validator(val) {
      if (!val) return true;
      return ["auto", "fixed"].includes(val);
    }
  },
  verticalAlign: {
    type: String,
    default: "middle",
    validator: function validator(val) {
      if (!val) return true;
      return ["top", "middle", "bottom"].includes(val);
    }
  },
  onCellClick: Function,
  onRowClick: Function,
  onScroll: Function
};

export { baseTableProps as default };
