// tdesign-mobile-vue/esm/tabs/props.js
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

var props = {
  animation: {
    type: Object
  },
  bottomLineMode: {
    type: String,
    default: "fixed",
    validator: function validator(val) {
      if (!val) return true;
      return ["fixed", "auto", "full"].includes(val);
    }
  },
  list: {
    type: Array
  },
  showBottomLine: {
    type: Boolean,
    default: true
  },
  size: {
    type: String,
    default: "medium",
    validator: function validator(val) {
      if (!val) return true;
      return ["medium", "large"].includes(val);
    }
  },
  spaceEvenly: {
    type: Boolean,
    default: true
  },
  sticky: Boolean,
  stickyProps: {
    type: Object
  },
  swipeable: {
    type: Boolean,
    default: true
  },
  theme: {
    type: String,
    default: "line",
    validator: function validator(val) {
      if (!val) return true;
      return ["line", "tag", "card"].includes(val);
    }
  },
  value: {
    type: [String, Number],
    default: void 0
  },
  modelValue: {
    type: [String, Number],
    default: void 0
  },
  defaultValue: {
    type: [String, Number]
  },
  onChange: Function,
  onClick: Function,
  onScroll: Function
};

export { props as default };
