// tdesign-mobile-vue/es/dialog/props.mjs
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

var props = {
  actions: {
    type: [Array, Function]
  },
  buttonLayout: {
    type: String,
    default: "horizontal",
    validator: function validator(val) {
      if (!val) return true;
      return ["horizontal", "vertical"].includes(val);
    }
  },
  cancelBtn: {
    type: [String, Object, Function]
  },
  closeBtn: {
    type: Boolean,
    default: false
  },
  closeOnOverlayClick: {
    type: Boolean,
    default: void 0
  },
  confirmBtn: {
    type: [String, Object, Function]
  },
  content: {
    type: [String, Function]
  },
  destroyOnClose: Boolean,
  overlayProps: {
    type: Object,
    default: function _default() {
      return {};
    }
  },
  preventScrollThrough: {
    type: Boolean,
    default: true
  },
  showOverlay: {
    type: Boolean,
    default: true
  },
  title: {
    type: [String, Function]
  },
  visible: Boolean,
  width: {
    type: [String, Number]
  },
  zIndex: {
    type: Number
  },
  onCancel: Function,
  onClose: Function,
  onClosed: Function,
  onConfirm: Function,
  onOverlayClick: Function
};

export { props as default };
