// tdesign-mobile-vue/lib/shared/useTest/index.js
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

import _defineProperty from '@babel/runtime/helpers/defineProperty';
import _asyncToGenerator from '@babel/runtime/helpers/asyncToGenerator';
import _regeneratorRuntime from '@babel/runtime/regenerator';
import { sleep } from '../util.js';
import '@babel/runtime/helpers/slicedToArray';
import 'lodash/isNumber';
import '../../config.js';

function useTestUtils() {
  var makeScroll = /*#__PURE__*/function () {
    var _ref = _asyncToGenerator(/*#__PURE__*/_regeneratorRuntime.mark(function _callee(dom, name, offset) {
      var eventTarget, evt;
      return _regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) switch (_context.prev = _context.next) {
          case 0:
            eventTarget = dom === document.documentElement ? window : dom;
            dom[name] = offset;
            evt = new CustomEvent("scroll", {
              detail: {
                target: _defineProperty({}, name, offset)
              }
            });
            eventTarget.dispatchEvent(evt);
            _context.next = 6;
            return sleep(0);
          case 6:
          case "end":
            return _context.stop();
        }
      }, _callee);
    }));
    return function makeScroll(_x, _x2, _x3) {
      return _ref.apply(this, arguments);
    };
  }();
  return {
    makeScroll: makeScroll,
    sleep: sleep
  };
}
function useVitest(vi) {
  var getBoundingClientRect = function getBoundingClientRect(dom) {
    return vi.spyOn(dom, "getBoundingClientRect");
  };
  return {
    getBoundingClientRect: getBoundingClientRect
  };
}

export { useTestUtils, useVitest };
