// tdesign-mobile-vue/es/search/props.mjs
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

var searchProps = {
  action: {
    type: [String, Function],
    default: ""
  },
  autocompleteOptions: {
    type: Array
  },
  center: Boolean,
  clearable: {
    type: Boolean,
    default: true
  },
  disabled: Boolean,
  focus: Boolean,
  leftIcon: {
    type: [String, Function],
    default: "search"
  },
  placeholder: {
    type: String,
    default: ""
  },
  readonly: Boolean,
  shape: {
    type: String,
    default: "square",
    validator: function validator(val) {
      if (!val) return true;
      return ["square", "round"].includes(val);
    }
  },
  value: {
    type: String,
    default: void 0
  },
  modelValue: {
    type: String,
    default: void 0
  },
  defaultValue: {
    type: String,
    default: ""
  },
  onActionClick: Function,
  onBlur: Function,
  onChange: Function,
  onClear: Function,
  onFocus: Function,
  onSearch: Function,
  onSubmit: Function
};

export { searchProps as default };
