// tdesign-mobile-vue/cjs/action-sheet/props.js
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var props = {
  align: {
    type: String,
    default: "center",
    validator: function validator(val) {
      if (!val) return true;
      return ["center", "left"].includes(val);
    }
  },
  cancelText: {
    type: String,
    default: ""
  },
  count: {
    type: Number,
    default: 8
  },
  description: {
    type: String,
    default: ""
  },
  items: {
    type: Array,
    default: [],
    required: true
  },
  showCancel: {
    type: Boolean,
    default: true
  },
  theme: {
    type: String,
    default: "list",
    validator: function validator(val) {
      if (!val) return true;
      return ["list", "grid"].includes(val);
    }
  },
  visible: {
    type: Boolean,
    default: void 0
  },
  modelValue: {
    type: Boolean,
    default: void 0
  },
  defaultVisible: Boolean,
  onCancel: Function,
  onClose: Function,
  onSelected: Function
};

exports["default"] = props;
