// tdesign-mobile-vue/es/shared/useTest/index.mjs
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

import { _ as _defineProperty } from '../../_chunks/dep-dc33ef46.mjs';
import { _ as _asyncToGenerator, r as regenerator } from '../../_chunks/dep-1f2e940b.mjs';
import { sleep } from '../util.mjs';
import '../../_chunks/dep-b617aa7d.mjs';
import '../../_chunks/dep-57a75b98.mjs';
import '../../_chunks/dep-4c2819a4.mjs';
import '../../_chunks/dep-4b5beaed.mjs';
import '../../_chunks/dep-5030b0d2.mjs';
import '../../_chunks/dep-c6268b4a.mjs';
import '../../_chunks/dep-06d482c6.mjs';
import '../../config.mjs';

function useTestUtils() {
  var makeScroll = /*#__PURE__*/function () {
    var _ref = _asyncToGenerator(/*#__PURE__*/regenerator.mark(function _callee(dom, name, offset) {
      var eventTarget, evt;
      return regenerator.wrap(function _callee$(_context) {
        while (1) switch (_context.prev = _context.next) {
          case 0:
            eventTarget = dom === document.documentElement ? window : dom;
            dom[name] = offset;
            evt = new CustomEvent("scroll", {
              detail: {
                target: _defineProperty({}, name, offset)
              }
            });
            eventTarget.dispatchEvent(evt);
            _context.next = 6;
            return sleep(0);
          case 6:
          case "end":
            return _context.stop();
        }
      }, _callee);
    }));
    return function makeScroll(_x, _x2, _x3) {
      return _ref.apply(this, arguments);
    };
  }();
  return {
    makeScroll: makeScroll,
    sleep: sleep
  };
}
function useVitest(vi) {
  var getBoundingClientRect = function getBoundingClientRect(dom) {
    return vi.spyOn(dom, "getBoundingClientRect");
  };
  return {
    getBoundingClientRect: getBoundingClientRect
  };
}

export { useTestUtils, useVitest };
