// tdesign-mobile-vue/lib/hooks/useLockScroll.js
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

import { onMounted, nextTick, onActivated, onDeactivated, onBeforeUnmount, watch } from 'vue';
import { useTouch } from '../_util/useTouch.js';
import getScrollParent from '../_util/getScrollParent.js';
import { supportsPassive } from '../_util/supportsPassive.js';
import '../shared/util.js';
import '@babel/runtime/helpers/slicedToArray';
import 'lodash/isNumber';
import '../config.js';

var totalLockCount = 0;
var mounted = null;
function useLockScroll(rootRef, shouldLock, componentName) {
  var touch = useTouch();
  var BODY_LOCK_CLASS = "".concat(componentName, "--lock");
  var onTouchMove = function onTouchMove(event) {
    touch.move(event);
    var direction = touch.deltaY.value > 0 ? "10" : "01";
    var el = getScrollParent(event.target, rootRef.value);
    if (!el) return;
    var scrollHeight = el.scrollHeight,
      offsetHeight = el.offsetHeight,
      scrollTop = el.scrollTop;
    var status = "11";
    if (scrollTop === 0) {
      status = offsetHeight >= scrollHeight ? "00" : "01";
    } else if (scrollTop + offsetHeight >= scrollHeight) {
      status = "10";
    }
    if (status !== "11" && touch.isVertical() && !(parseInt(status, 2) & parseInt(direction, 2))) {
      if (event.cancelable) {
        event.preventDefault();
      }
    }
  };
  var lock = function lock() {
    document.addEventListener("touchstart", touch.start);
    document.addEventListener("touchmove", onTouchMove, supportsPassive.value ? {
      passive: false
    } : false);
    if (!totalLockCount) {
      document.body.classList.add(BODY_LOCK_CLASS);
    }
    totalLockCount += 1;
  };
  var unlock = function unlock() {
    if (totalLockCount) {
      document.removeEventListener("touchstart", touch.start);
      document.removeEventListener("touchmove", onTouchMove);
      totalLockCount -= 1;
      if (!totalLockCount) {
        document.body.classList.remove(BODY_LOCK_CLASS);
      }
    }
  };
  var init = function init() {
    return shouldLock() && lock();
  };
  var destroy = function destroy() {
    return shouldLock() && unlock();
  };
  onMounted(function () {
    init();
    nextTick(function () {
      mounted = true;
    });
  });
  onActivated(function () {
    if (mounted) {
      init();
    }
  });
  onDeactivated(destroy);
  onBeforeUnmount(destroy);
  watch(shouldLock, function (value) {
    value ? lock() : unlock();
  });
}

export { useLockScroll };
