// tdesign-mobile-vue/cjs/cascader/props.js
/**
 * tdesign v1.5.0
 * (c) 2024 TDesign Group
 * @license MIT
 */

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var props = {
  checkStrictly: Boolean,
  closeBtn: {
    type: [Boolean, Function],
    default: true
  },
  keys: {
    type: Object
  },
  options: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  placeholder: {
    type: [String, Function],
    default: "\u9009\u62E9\u9009\u9879"
  },
  subTitles: {
    type: Array,
    default: function _default() {
      return [];
    }
  },
  theme: {
    type: String,
    default: "step",
    validator: function validator(val) {
      if (!val) return true;
      return ["step", "tab"].includes(val);
    }
  },
  title: {
    type: [String, Function]
  },
  value: {
    type: [String, Number],
    default: void 0
  },
  modelValue: {
    type: [String, Number],
    default: void 0
  },
  defaultValue: {
    type: [String, Number]
  },
  visible: Boolean,
  onChange: Function,
  onClose: Function,
  onPick: Function
};

exports["default"] = props;
